lib/internal/util/embedding.js

```diff
@@ -15,7 +16,7 @@ const { Module, wrapSafe } = require('internal/modules/cjs/loader');
 
 function embedderRunCjs(contents) {
   const filename = process.execPath;
```

Comment:
I think this means once the file gets moved to a different path, the code cache would get invalidated. We probably need to fix the entry point file name somehow (e.g. only use the file name and ignore the directories) when code cache is used.

Reply:
The `cachedDataRejected` assertion in `lib/internal/modules/cjs/loader.js` wasn't failing. Maybe it's a bug similar to https://github.com/nodejs/node/issues/48175? Anyways, I've updated the code to preserve the name of the main script, PTAL.